### PR TITLE
New Rules Flyout, toggle btn and isRequired field for formLable

### PR DESCRIPTION
**Pull request by isaac73**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

...

**Checklist:**

- [ ] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: public/locales/en/translations.json [hunk 1]
```diff
@@ -42,6 +42,27 @@
...
+      "condition": "Condition",
+      "field": "Field",
+      "calculation": "Calculation",
+      "timePeriod": "Time Period",
+      "operator": "Operator",
+      "value": "Value"
+    },
+    "severityLevel": "Severity level",
+    "ruleStatus": "Rule status",
+    "devicesAffected": "devices affected by this rule",
```
Comment (ppathan) [inline public/locales/en/translations.json]: nit: Devices?

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 2]
```diff
@@ -33,16 +47,25 @@ export class RuleNew extends LinkedComponent {
     this.state = {
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: There is a TODO on line 41 that can be removed.

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 3]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
   }
 
   render() {
-    const { onClose } = this.props;
-
+    const { onClose, t } = this.props;
+    const name = this.ruleNameLink.forkTo('name')
+        .withValidator(ruleNameValidator);
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: nit: extra indent. I think we do 2 spaces

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 4]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
-                  type="textarea"
-                  placeholder="Description"
-                  link={this.description} />
-              </FormGroup>
+                <FormGroup>
+                  <FormLabel isRequired="true">{t(`rulesFlyout.ruleName`)}</FormLabel>
+                  <FormControl
+                    type="text"
+                    className="long"
+                    placeholder="Rule name"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 5]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                    link={name} />
+                </FormGroup>
+                <FormGroup>
+                  <FormLabel>{t(`rulesFlyout.description`)}</FormLabel>
+                  <FormControl
+                    type="textarea"
+                    placeholder="Description"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 6]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                    link={this.descriptionLink} />
+                </FormGroup>
+                <FormGroup>
+                  <FormLabel isRequired="true">{t(`rulesFlyout.deviceGroup`)}</FormLabel>
+                  <FormControl
+                    type="select"
+                    className="long"
+                    placeholder="DeviceGroup"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 7]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                    {
+                      conditionLinks.length > 1 &&
+                      <Btn svg={svgs.trash} onClick={this.deleteCondition(idx)}>Delete</Btn>
+                    }
+                    <FormGroup>
+                      <FormLabel isRequired="true">{t(`rulesFlyout.condition.field`)}</FormLabel>
+                      <FormControl
+                        type="select"
+                        className="long"
+                        placeholder="Select field"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 8]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                        link={condition.field} />
+                    </FormGroup>
+                    <FormGroup>
+                      <FormLabel isRequired="true">{t(`rulesFlyout.condition.calculation`)}</FormLabel>
+                      <FormControl
+                        type="select"
+                        className="long"
+                        placeholder="Select"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 9]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                        link={condition.calculation} />
+                    </FormGroup>
+                    <FormGroup>
+                      <FormLabel isRequired="true">{t(`rulesFlyout.condition.timePeriod`)}</FormLabel>
+                      <FormControl
+                        type="duration"
+                        className="long"
+                        placeholder="Select"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 10]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                        link={condition.timePeriod} />
+                    </FormGroup>
+                    <FormGroup>
+                      <FormLabel isRequired="true">{t(`rulesFlyout.condition.operator`)}</FormLabel>
+                      <FormControl
+                        type="select"
+                        className="short"
+                        placeholder="Select operator"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 11]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                        link={condition.operator} />
+                    </FormGroup>
+                    <FormGroup>
+                      <FormLabel isRequired="true">{t(`rulesFlyout.condition.value`)}</FormLabel>
+                      <FormControl
+                        type="text"
+                        placeholder="Enter value"
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: Add this to translation.js

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js [hunk 12]
```diff
@@ -53,69 +76,144 @@ export class RuleNew extends LinkedComponent {
...
+                </FormGroup>
               </Section.Content>
             </Section.Container>
 
             <Section.Container collapsable={false}>
-              <Section.Header>Conditions</Section.Header>
+              <Section.Header>{t(`rulesFlyout.conditions`)}</Section.Header>
               <Section.Content>
-                <Btn svg={svgs.plus} onClick={this.addCondition}>Add condition</Btn>
+                <Btn svg={svgs.plus} onClick={this.addCondition}>{`rulesFlyout.addCondition`}</Btn>
```
Comment (molly-moen) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: t(`rulesFlyout...`)

File: src/components/shared/forms/styles/toggleBtn.scss [hunk 13]
```diff
@@ -33,4 +36,25 @@
 
     .disabled { stroke: themed('colorDisabledLine'); }
   }
+  .input-contents {
+    display: flex;
+    flex-flow: column nowrap;
+    flex-grow: 1;
+    @include rem-fallback(margin-left, 10px);
+
+    > .form-group-label {
```
Comment (molly-moen) [inline src/components/shared/forms/styles/toggleBtn.scss]: fix indendentation

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.scss [hunk 14]
```diff
@@ -1,9 +1,41 @@
...
     margin-top: 20px;
   }
+
+  .devices-affected{
+    display: flex;
+    flex-flow: row nowrap;
+    position: relative;
+    top: 18.5px;
+    height: 67px;
+  }
```
Comment (molly-moen) [inline src/components/pages/rules/flyouts/ruleNew/ruleNew.scss]: newline

File: public/locales/en/translations.json (hunk 1 above)
Comment (isaac73) [reply public/locales/en/translations.json]: Is that wrong?

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 2 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 3 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 5 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 7 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 8 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 9 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 10 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 11 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.js (hunk 12 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.js]: fixed

File: src/components/shared/forms/styles/toggleBtn.scss (hunk 13 above)
Comment (isaac73) [reply src/components/shared/forms/styles/toggleBtn.scss]: looks correct to me!

File: src/components/pages/rules/flyouts/ruleNew/ruleNew.scss (hunk 14 above)
Comment (isaac73) [reply src/components/pages/rules/flyouts/ruleNew/ruleNew.scss]: fixed

File: src/components/shared/forms/styles/toggleBtn.scss (hunk 13 above)
Comment (molly-moen) [reply src/components/shared/forms/styles/toggleBtn.scss]: nvm my mistake

File: public/locales/en/translations.json [hunk 15]
```diff
@@ -45,17 +45,24 @@
...
     "condition": {
       "condition": "Condition",
       "field": "Field",
+      "fieldPlaceholder": "Select field",
       "calculation": "Calculation",
+      "calculationPlaceholder": "select",
       "timePeriod": "Time Period",
       "operator": "Operator",
-      "value": "Value"
+      "operatorPlaceholder": "> =",
```
Comment (ppathan) [inline public/locales/en/translations.json]: It should be `>=`

Comment (isaac73) [reply public/locales/en/translations.json]: fixed

**Pull request merged**